src/mpid/ch4/netmod/ofi/ofi_types.h

```diff
@@ -397,6 +350,23 @@ typedef struct {
     char kvsname[MPIDI_KVSAPPSTRLEN];
     char pname[MPI_MAX_PROCESSOR_NAME];
     int port_name_tag_mask[MPIR_MAX_CONTEXT_MASK];
+
+    /* Capability settings */
+#ifdef MPIDI_OFI_ENABLE_RUNTIME_CHECKS
```

Comment:
So your suggestion is something like this?

```
/* Get the available providers and default capabilities */
caps = NULL;
fi_info(..., &caps, ...);

/* Query again with the capability that I care about + the original defaults */
caps[0] |= /* capability that I care about */;
fi_info(..., &caps, ...);
```

Without diving in to things to deeply, that could probably work. I'd have to start playing with it to know for sure.